packages/navigation/src/UILargeTitleHeader/useOnScrollHandler/onScroll.native.tsx

```diff
@@ -40,15 +42,14 @@ export default function (
         }
 
         /**
-         * It can be undefined on Android and this is fine
-         * since the guard is needed only for iOS
+         * The fix is needed only for iOS
          *
          * On iOS `onScroll` event could fire on mount sometimes,
          * that's likely a bug in RN or iOS itself.
          * To prevent changes when there wasn't onBeginDrag event
          * (so it's likely not an actual scroll) using a guard
          */
-        if (ctx != null && !ctx.scrollTouchGuard) {
+        if (isIOS && !ctx != null && !ctx.scrollTouchGuard) {
```

Comment:
Maybe you wanted to do this? I'm not sure why have you added the leading `!` to `ctx != null` condition...
```suggestion
        if (isIOS && ctx != null && !ctx.scrollTouchGuard) {
```